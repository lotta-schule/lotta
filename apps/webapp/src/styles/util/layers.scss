@layer theme, lotta, hubert, third-party;
