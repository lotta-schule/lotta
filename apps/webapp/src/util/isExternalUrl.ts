/**
 * Checks if a URL is external (not relative or on the same host).
 * @param url - The URL to check
 * @returns true if the URL is external, false otherwise
 */
export const isExternalUrl = (url: string): boolean => {
  // Relative URLs starting with / are internal
  if (url.startsWith('/')) {
    return false;
  }

  // Check if it's an absolute URL
  try {
    new URL(url);
    // If we can create a URL object, it's an external absolute URL
    return true;
  } catch {
    // If URL parsing fails, treat it as an internal relative URL
    return false;
  }
};
