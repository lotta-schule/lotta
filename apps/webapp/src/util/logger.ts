import * as Sentry from '@sentry/nextjs';

type LogLevel = 'debug' | 'info' | 'warning' | 'error';

type LogContext = {
  [key: string]: any;
};

const addTraceContext = (context: LogContext = {}): LogContext => {
  const span = Sentry.getActiveSpan();
  if (span) {
    const { traceId, spanId } = span.spanContext();
    return {
      ...context,
      traceId,
      spanId,
    };
  }
  return context;
};

const log = (level: LogLevel, message: string, context?: LogContext) => {
  const enrichedContext = addTraceContext(context);

  Sentry.addBreadcrumb({
    level,
    message,
    data: enrichedContext,
  });

  const fnName = level === 'warning' ? 'warn' : level;

  console[fnName](message, enrichedContext);
};

export const Logger = {
  debug: (message: string, context?: LogContext) =>
    log('debug', message, context),
  info: (message: string, context?: LogContext) =>
    log('info', message, context),
  warn: (message: string, context?: LogContext) =>
    log('warning', message, context),
  error: (message: string, context?: LogContext) =>
    log('error', message, context),
};
