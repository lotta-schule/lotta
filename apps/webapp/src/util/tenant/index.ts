export * from './useTenant';
export * from './useUserGroups';
