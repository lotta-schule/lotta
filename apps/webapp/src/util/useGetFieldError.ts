import * as React from 'react';
import { ApolloError } from '@apollo/client';

export const useGetFieldError = (error?: ApolloError) => {
  return React.useCallback(
    (fieldName: string): string | false => {
      if (error) {
        const messages: string[] | undefined =
          error.graphQLErrors &&
          (error.graphQLErrors.map(
            (graphQLError: any) =>
              graphQLError &&
              graphQLError.details &&
              graphQLError.details[fieldName]
          ) as string[] | undefined);
        if (messages) {
          return messages[0];
        }
      }
      return false;
    },
    [error]
  );
};
