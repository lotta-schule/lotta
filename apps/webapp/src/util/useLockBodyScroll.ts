import * as React from 'react';

export const useLockBodyScroll = (): void => {
  React.useLayoutEffect(() => {
    const originalStyle = window.getComputedStyle(document.body).overflow;
    document.body.style.overflow = 'hidden';
    return () => {
      document.body.style.overflow = originalStyle;
    };
  }, []);
};
