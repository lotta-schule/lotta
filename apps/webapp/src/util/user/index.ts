export * from './useCurrentUser';
export * from './useOnLogout';
