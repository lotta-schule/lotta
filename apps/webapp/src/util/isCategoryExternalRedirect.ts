import { CategoryModel } from 'model/CategoryModel';
import { isExternalUrl } from './isExternalUrl';

/**
 * Checks if a category has an external redirect URL.
 * @param category - The category to check
 * @returns true if the category has an external redirect, false otherwise
 */
export const isCategoryExternalRedirect = (
  category: CategoryModel
): boolean => {
  return !!category.redirect && isExternalUrl(category.redirect);
};
