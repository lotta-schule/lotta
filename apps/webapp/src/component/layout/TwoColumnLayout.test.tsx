import { render } from 'test/util';
import {
  TwoColumnLayout,
  TwoColumnLayoutContent,
  TwoColumnLayoutSidebar,
} from './TwoColumnLayout';

describe('TwoColumnLayout', () => {
  it('should render successfully', () => {
    const screen = render(
      <TwoColumnLayout>
        <TwoColumnLayoutSidebar>Sidebar</TwoColumnLayoutSidebar>
        <TwoColumnLayoutContent>Content</TwoColumnLayoutContent>
      </TwoColumnLayout>
    );

    expect(screen.container.firstChild).toMatchSnapshot();
  });
});
