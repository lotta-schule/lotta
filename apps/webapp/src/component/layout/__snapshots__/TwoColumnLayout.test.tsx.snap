// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TwoColumnLayout > should render successfully 1`] = `
<div
  data-overlay-container="true"
>
  <div
    class="_root_a3c7d4"
  >
    <aside
      class="_sidebar_a3c7d4"
    >
      Sidebar
    </aside>
    <main
      class="_main_a3c7d4"
    >
      Content
    </main>
  </div>
</div>
`;
